planning/behavior_path_lane_change_module/src/scene.cpp

```diff
@@ -856,23 +856,38 @@ LaneChangeTargetObjects NormalLaneChange::getTargetObjects(
   target_objects.other_lane.reserve(target_obj_index.other_lane.size());
 
   // objects in current lane
+
+  lanelet::ConstLanelet current_lane;
+  if (getRouteHandler()->getClosestLaneletWithinRoute(getEgoPose(), &current_lane)) {
+  }
+  const auto ego_footprint =
+    utils::lane_change::getEgoCurrentFootprint(getEgoPose(), common_parameters.vehicle_info);
+  const auto is_ego_within_intersection =
+    utils::lane_change::isWithinIntersection(getRouteHandler(), current_lane, ego_footprint);
+
+  const auto check_prepare_phase =
+    is_ego_within_intersection || lane_change_parameters_->enable_prepare_segment_collision_check;
```

Comment:
I could not clearly understand even when `enable_prepare_segment_collision_check` is false but `is_ego_within_intersection` is true, the prepare phase is checked.

Reply:
@shmpwk 
Sorry for the late reply.
`check_prepare_phase` only true when ego is in intersection. Otherwise it should be false.